// yadda/examples/component-browser/build/build.js

/**
 * Require the given path.
 *
 * @param {String} path
 * @return {Object} exports
 * @api public
 */

function require(path, parent, orig) {
  var resolved = require.resolve(path);

  // lookup failed
  if (null == resolved) {
    orig = orig || path;
    parent = parent || 'root';
    var err = new Error('Failed to require "' + orig + '" from "' + parent + '"');
    err.path = orig;
    err.parent = parent;
    err.require = true;
    throw err;
  }

  var module = require.modules[resolved];

  // perform real require()
  // by invoking the module's
  // registered function
  if (!module._resolving && !module.exports) {
    var mod = {};
    mod.exports = {};
    mod.client = mod.component = true;
    module._resolving = true;
    module.call(this, mod.exports, require.relative(resolved), mod);
    delete module._resolving;
    module.exports = mod.exports;
  }

  return module.exports;
}

/**
 * Registered modules.
 */

require.modules = {};

/**
 * Registered aliases.
 */

require.aliases = {};

/**
 * Resolve `path`.
 *
 * Lookup:
 *
 *   - PATH/index.js
 *   - PATH.js
 *   - PATH
 *
 * @param {String} path
 * @return {String} path or null
 * @api private
 */

require.resolve = function(path) {
  if (path.charAt(0) === '/') path = path.slice(1);

  var paths = [
    path,
    path + '.js',
    path + '.json',
    path + '/index.js',
    path + '/index.json'
  ];

  for (var i = 0; i < paths.length; i++) {
    var path = paths[i];
    if (require.modules.hasOwnProperty(path)) return path;
    if (require.aliases.hasOwnProperty(path)) return require.aliases[path];
  }
};

/**
 * Normalize `path` relative to the current path.
 *
 * @param {String} curr
 * @param {String} path
 * @return {String}
 * @api private
 */

require.normalize = function(curr, path) {
  var segs = [];

  if ('.' != path.charAt(0)) return path;

  curr = curr.split('/');
  path = path.split('/');

  for (var i = 0; i < path.length; ++i) {
    if ('..' == path[i]) {
      curr.pop();
    } else if ('.' != path[i] && '' != path[i]) {
      segs.push(path[i]);
    }
  }

  return curr.concat(segs).join('/');
};

/**
 * Register module at `path` with callback `definition`.
 *
 * @param {String} path
 * @param {Function} definition
 * @api private
 */

require.register = function(path, definition) {
  require.modules[path] = definition;
};

/**
 * Alias a module definition.
 *
 * @param {String} from
 * @param {String} to
 * @api private
 */

require.alias = function(from, to) {
  if (!require.modules.hasOwnProperty(from)) {
    throw new Error('Failed to alias "' + from + '", it does not exist');
  }
  require.aliases[to] = from;
};

/**
 * Return a require function relative to the `parent` path.
 *
 * @param {String} parent
 * @return {Function}
 * @api private
 */

require.relative = function(parent) {
  var p = require.normalize(parent, '..');

  /**
   * lastIndexOf helper.
   */

  function lastIndexOf(arr, obj) {
    var i = arr.length;
    while (i--) {
      if (arr[i] === obj) return i;
    }
    return -1;
  }

  /**
   * The relative require() itself.
   */

  function localRequire(path) {
    var resolved = localRequire.resolve(path);
    return require(resolved, parent, path);
  }

  /**
   * Resolve relative to the parent.
   */

  localRequire.resolve = function(path) {
    var c = path.charAt(0);
    if ('/' == c) return path.slice(1);
    if ('.' == c) return require.normalize(p, path);

    // resolve deps by returning
    // the dep in the nearest "deps"
    // directory
    var segs = parent.split('/');
    var i = lastIndexOf(segs, 'deps') + 1;
    if (!i) i = 0;
    path = segs.slice(0, i + 1).join('/') + '/deps/' + path;
    return path;
  };

  /**
   * Check if module is defined at `path`.
   */

  localRequire.exists = function(path) {
    return require.modules.hasOwnProperty(localRequire.resolve(path));
  };

  return localRequire;
};
require.register("acuminous-yadda/lib/Array.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var fn = require('./fn');\n\
\n\
module.exports = function(obj) {\n\
\n\
    function ensure_array(obj) {\n\
        var array = obj ? [].concat(obj) : [];\n\
        array.in_array = fn.curry(array, in_array, array);\n\
        array.each = fn.curry(array, each, array);\n\
        array.each_async = fn.curry(array, each_async, array);\n\
        array.collect = fn.curry(array, collect, array);\n\
        array.flatten = fn.curry(array, flatten, array);\n\
        array.inject = fn.curry(array, inject, array);\n\
        array.push_all = fn.curry(array, push_all, array);\n\
        array.find_all = fn.curry(array, find_all, array);\n\
        array.find = fn.curry(array, find, array);\n\
        array.naked = fn.curry(array, naked, array);\n\
        return array;\n\
    }\n\
\n\
    function is_array(obj) {\n\
        return Object.prototype.toString.call(obj) === '[object Array]';\n\
    }\n\
\n\
    function in_array(items, item) {\n\
        for (var i = 0; i < items.length; i++) {\n\
            if (items[i] == item) {\n\
                return true;\n\
            }\n\
        }\n\
    }\n\
\n\
    function flatten(items) {\n\
        if (!is_array(items)) return [items];\n\
        if (items.length === 0) return items;\n\
        var head = flatten(items[0]);\n\
        var tail = flatten(items.slice(1));\n\
        return ensure_array(head.concat(tail));\n\
    }\n\
\n\
    function each(items, iterator) {\n\
        var result;\n\
        for (var i = 0; i < items.length; i++) {\n\
            result = iterator(items[i], i);\n\
        }\n\
        return result;\n\
    }\n\
\n\
    function each_async(items, iterator, callback) {\n\
        callback = callback || fn.noop;\n\
        if (!items.length) return callback();\n\
        var completed = 0;\n\
        var iterate = function() {\n\
            iterator(items[completed], completed, function(err, result) {\n\
                if (err) return callback(err);\n\
                if (++completed >= items.length) return callback(null, result);\n\
                iterate();\n\
            });\n\
        };\n\
        iterate();\n\
    }\n\
\n\
    function collect(items, iterator) {\n\
        var results = ensure_array();\n\
        for (var i = 0; i < items.length; i++) {\n\
            results.push(iterator(items[i]));\n\
        }\n\
        return results;\n\
    }\n\
\n\
    function inject(items, default_value, iterator) {\n\
        var result = default_value;\n\
        for (var i = 0; i < items.length; i++) {\n\
            result = iterator(result, items[i]);\n\
        }\n\
        return result;\n\
    }\n\
\n\
    function push_all(items, more_items) {\n\
        more_items = more_items ? [].concat(more_items) : [];\n\
        for (var i = 0; i < more_items.length; i++) {\n\
            items.push(more_items[i]);\n\
        }\n\
    }\n\
\n\
    function find_all(items, test) {\n\
        var results = ensure_array();\n\
        for (var i = 0; i < items.length; i++) {\n\
            if (test(items[i])) {\n\
                results.push(items[i]);\n\
            }\n\
        }\n\
        return results;\n\
    }\n\
\n\
    function find(items, test) {\n\
        var result;\n\
        for (var i = 0; i < items.length; i++) {\n\
            if (test(items[i])) {\n\
                result = items[i];\n\
                break;\n\
            }\n\
        }\n\
        return result;\n\
    }\n\
\n\
    function naked(items) {\n\
        return [].concat(items);\n\
    }\n\
\n\
    return ensure_array(obj);\n\
};\n\
));
require.register("acuminous-yadda/lib/Competition.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var LevenshteinDistanceScore = require('./LevenshteinDistanceScore');\n\
var $ = require('./Array');\n\
\n\
// Understands appropriateness of macros in relation to a specific step\n\
var Competition = function(step, macros) {\n\
\n\
    var results = [];\n\
\n\
    this.validate = function() {\n\
        if (is_undefined()) return { step: step, valid: false, reason: 'Undefined Step' };\n\
        if (is_ambiguous()) return { step: step, valid: false, reason: 'Ambiguous Step (Patterns [' + winning_patterns() + '] are all equally good candidates)' };\n\
        return { step: step, valid: true };\n\
    };\n\
\n\
    this.clear_winner = function() {\n\
        if (is_undefined()) throw new Error('Undefined Step: [' + step + ']');\n\
        if (is_ambiguous()) throw new Error('Ambiguous Step: [' + step + ']. Patterns [' + winning_patterns() + '] match equally well.');\n\
        return this.winner();\n\
    };\n\
\n\
    function is_undefined() {\n\
        return results.length === 0;\n\
    }\n\
\n\
    function is_ambiguous() {\n\
        return (results.length > 1) && results[0].score.equals(results[1].score);\n\
    }\n\
\n\
    this.winner = function() {\n\
        return results[0].macro;\n\
    };\n\
\n\
    function winning_patterns() {\n\
        return results.find_all(by_winning_score).collect(macro_signatures).join(', ');\n\
    }\n\
\n\
    function rank(step, macros) {\n\
        results = macros.collect(function(macro) {\n\
            return {\n\
                macro: macro,\n\
                score: new LevenshteinDistanceScore(step, macro.levenshtein_signature())\n\
            };\n\
        }).sort( by_ascending_score );\n\
    }\n\
\n\
    function by_ascending_score(a, b) {\n\
        return b.score.beats(a.score);\n\
    }\n\
\n\
    function by_winning_score(result) {\n\
        return result.score.equals(results[0].score);\n\
    }\n\
\n\
    function macro_signatures(result) {\n\
        return result.macro.toString();\n\
    }\n\
\n\
    rank(step, $(macros));\n\
};\n\
\n\
module.exports = Competition;\n\
));
require.register("acuminous-yadda/lib/Context.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
// Constructs an object that macros will be bound to before execution\n\
var Context = function(properties) {\n\
\n\
    // I was previously getting some weird errors using instanceof to determine if\n\
    // the \"other\" object was a context object. Using pTFUHht733hM6wfnruGLgAu6Uqvy7MVp instead\n\
    this.pTFUHht733hM6wfnruGLgAu6Uqvy7MVp = true;\n\
    this.properties = {};\n\
\n\
    this.merge = function(other) {\n\
        if (other && other.pTFUHht733hM6wfnruGLgAu6Uqvy7MVp) return this.merge(other.properties);\n\
        return new Context(this.properties)._merge(other);\n\
    };\n\
\n\
    this._merge = function(other) {\n\
        for (var key in other) { this.properties[key] = other[key]; }\n\
        return this;\n\
    };\n\
\n\
    this._merge(properties);\n\
};\n\
\n\
module.exports = Context;\n\
));
require.register("acuminous-yadda/lib/Dictionary.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var $ = require('./Array');\n\
var RegularExpression = require('./RegularExpression');\n\
\n\
// Understands term definitions\n\
var Dictionary = function(prefix) {\n\
\n\
    /* jslint shadow: true */\n\
    var prefix = prefix || '$';\n\
    var terms = {};\n\
    var term_pattern = new RegularExpression(new RegExp('(?:^|[^\\\\\\\\])\\\\' + prefix + '(\\\\w+)', 'g'));\n\
    var _this = this;\n\
\n\
    this.define = function(term, definition) {\n\
        if (this.is_defined(term)) throw new Error('Duplicate definition: [' + term + ']');\n\
        terms[term] = normalise(definition);\n\
        return this;\n\
    };\n\
\n\
    this.is_defined = function(term) {\n\
        return terms[term];\n\
    };\n\
\n\
    this.expand = function(term, already_expanding) {\n\
        if (!is_expandable(term)) return term;\n\
        return expand_sub_terms(term, $(already_expanding));\n\
    };\n\
\n\
    this.merge = function(other) {\n\
        if (other._prefix() != this._prefix()) throw new Error('Cannot merge dictionaries with different prefixes');\n\
        return new Dictionary(prefix)._merge(this)._merge(other);\n\
    };\n\
\n\
    this._merge = function(other) {\n\
        other.each_term(this.define.bind(this));\n\
        return this;\n\
    };\n\
\n\
    this._prefix = function() {\n\
        return prefix;\n\
    };\n\
\n\
    this.each_term = function(callback) {\n\
        for (var key in terms) {\n\
            callback(key, terms[key]);\n\
        }\n\
    };\n\
\n\
    var expand_sub_terms = function(term, already_expanding) {\n\
        return get_sub_terms(term).each(function(sub_term) {\n\
            if (already_expanding.in_array(sub_term)) throw new Error('Circular Definition: [' + already_expanding.join(', ') + ']');\n\
            var sub_term_definition = expand_sub_term(sub_term, already_expanding);\n\
            term = term.replace(prefix + sub_term, sub_term_definition);\n\
            return term;\n\
        });\n\
    };\n\
\n\
    var get_sub_terms = function(term) {\n\
        return term_pattern.groups(term);\n\
    };\n\
\n\
    var expand_sub_term = function(sub_term, already_expanding) {\n\
        var definition = terms[sub_term] || '(.+)';\n\
        return is_expandable(definition) ? _this.expand(definition, already_expanding.concat(sub_term)) : definition;\n\
    };\n\
\n\
    var normalise = function(definition) {\n\
        return definition.toString().replace(/^\\/|\\/$/g, '');\n\
    };\n\
\n\
    var is_expandable = function(definition) {\n\
        return term_pattern.test(definition);\n\
    };\n\
};\n\
\n\
\n\
module.exports = Dictionary;\n\
));
require.register("acuminous-yadda/lib/EventBus.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var $ = require('./Array');\n\
var fn = require('./fn');\n\
var event_bus = new EventBus();\n\
\n\
// A communication channel between event emitters and event listeners\n\
function EventBus() {\n\
\n\
    var event_handlers = $();\n\
    var _this = this;\n\
\n\
    this.send = function(event_name, event_data, next) {\n\
        if (arguments.length == 1) return this.send(event_name, {});\n\
        if (arguments.length == 2 && fn.is_function(event_data)) return this.send(event_name, {}, event_data);\n\
        notify_handlers(event_name, event_data);\n\
        next && next();\n\
        return this;\n\
    };\n\
\n\
    this.on = function(event_pattern, callback) {\n\
        event_handlers.push({ pattern: event_pattern, callback: callback });\n\
        return this;\n\
    };\n\
\n\
    var notify_handlers = function(event_name, event_data) {\n\
        find_handlers(event_name).each(function(callback) {\n\
            callback({ name: event_name, data: event_data });\n\
        });\n\
    };\n\
\n\
    var find_handlers = function(event_name) {\n\
        return event_handlers.find_all(function(handler) {\n\
            return new RegExp(handler.pattern).test(event_name);\n\
        }).collect(function(handler) {\n\
            return handler.callback;\n\
        });\n\
    };\n\
}\n\
\n\
function instance() {\n\
    return event_bus;\n\
}\n\
\n\
module.exports = {\n\
    instance: instance,\n\
    ON_SCENARIO: '__ON_SCENARIO__',\n\
    ON_STEP: '__ON_STEP__',\n\
    ON_EXECUTE: '__ON_EXECUTE__'\n\
};\n\
));
require.register("acuminous-yadda/lib/FileSearch.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var shims = require('./shims/index');\n\
var path = shims.path;\n\
var process = shims.process;\n\
var fs = shims.fs;\n\
var $ = require('./Array');\n\
\n\
// Searches for files in the given directories and their sub-directories, matching at least one of the given patterns\n\
var FileSearch = function(directories, patterns) {\n\
\n\
    /* jslint shadow: true */\n\
    var patterns = patterns || /.*/;\n\
\n\
    this.each = function(fn) {\n\
        this.list().forEach(fn);\n\
    };\n\
\n\
    this.list = function() {\n\
        return $(directories).inject($(), function(files, directory) {\n\
            return files.concat(list_files(directory).find_all(by_pattern));\n\
        });\n\
    };\n\
\n\
    var list_files = function(directory) {\n\
        return $(list_immediate_files(directory).concat(list_sub_directory_files(directory)));\n\
    };\n\
\n\
    var list_immediate_files = function(directory) {\n\
        return ls(directory).find_all(by_file);\n\
    };\n\
\n\
    var list_sub_directory_files = function(directory) {\n\
        return ls(directory).find_all(by_directory).inject($(), function(files, directory) {\n\
            return files.concat(list_files(directory));\n\
        });\n\
    };\n\
\n\
    var ls = function(directory) {\n\
        if (!fs.existsSync(directory)) return $();\n\
        return $(fs.readdirSync(directory)).collect(function(file) {\n\
            return path.join(directory, file);\n\
        });\n\
    };\n\
\n\
    var by_file = function(file) {\n\
        return !by_directory(file);\n\
    };\n\
\n\
    var by_directory = function(file) {\n\
        return fs.statSync(file).isDirectory();\n\
    };\n\
\n\
    var by_pattern = function(filename) {\n\
        return $(patterns).find(function(pattern) {\n\
            return new RegExp(pattern).test(filename);\n\
        });\n\
    };\n\
};\n\
\n\
module.exports = FileSearch;\n\
));
require.register("acuminous-yadda/lib/FeatureFileSearch.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var FileSearch = require('./FileSearch');\n\
\n\
var FeatureFileSearch = function(directories) {\n\
    this.constructor(directories, /.*\\.(?:feature|spec|specification)$/);\n\
};\n\
FeatureFileSearch.prototype = new FileSearch();\n\
\n\
module.exports = FeatureFileSearch;\n\
));
require.register("acuminous-yadda/lib/fn.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
module.exports = (function() {\n\
\n\
    var slice = Array.prototype.slice;\n\
\n\
    function curry(ctx, fn) {\n\
        var args = slice.call(arguments, 2);\n\
        return function() {\n\
            return fn.apply(ctx, args.concat(slice.call(arguments)));\n\
        };\n\
    }\n\
\n\
    function invoke(fn, ctx, args) {\n\
        return fn.apply(ctx, args);\n\
    }\n\
\n\
    function is_function(object) {\n\
        var getType = {};\n\
        return object && getType.toString.call(object) === '[object Function]';\n\
    }\n\
\n\
    function noop() {}\n\
\n\
    function asynchronize(ctx, fn) {\n\
        return function() {\n\
            var next = slice.call(arguments, arguments.length - 1)[0];\n\
            var args = slice.call(arguments, 0, arguments.length - 2);\n\
            fn.apply(ctx, args);\n\
            if (next) next();\n\
        };\n\
    }\n\
\n\
    return {\n\
        noop: noop,\n\
        async_noop: asynchronize(null, noop),\n\
        asynchronize: asynchronize,\n\
        is_function: is_function,\n\
        curry: curry,\n\
        invoke: invoke\n\
    };\n\
\n\
\n\
})();\n\
));
require.register("acuminous-yadda/lib/index.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var api = {\n\
    Yadda: require('./Yadda'),\n\
    EventBus: require('./EventBus'),\n\
    Interpreter: require('./Interpreter'),\n\
    Context: require('./Context'),\n\
    Library: require('./Library'),\n\
    Dictionary: require('./Dictionary'),\n\
    FeatureFileSearch: require('./FeatureFileSearch'),\n\
    FileSearch: require('./FileSearch'),\n\
    Platform: require('./Platform'),\n\
    localisation: require('./localisation/index'),\n\
    parsers: require('./parsers/index'),\n\
    plugins: require('./plugins/index'),\n\
    shims: require('./shims/index'),\n\
    createInstance: function() {\n\
        // Not everyone shares my sense of humour re the recursive api :(\n\
        // See https://github.com/acuminous/yadda/issues/111\n\
        return api.Yadda.apply(null, Array.prototype.slice.call(arguments, 0));\n\
    }\n\
};\n\
\n\
module.exports = api;\n\
));
require.register("acuminous-yadda/lib/Interpreter.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var Competition = require('./Competition');\n\
var Context = require('./Context');\n\
var EventBus = require('./EventBus');\n\
var $ = require('./Array');\n\
var fn = require('./fn');\n\
\n\
// Understands a scenario\n\
var Interpreter = function(libraries) {\n\
\n\
    /* jslint shadow: true */\n\
    var libraries = $(libraries);\n\
    var event_bus = EventBus.instance();\n\
    var _this = this;\n\
\n\
    this.requires = function(libraries) {\n\
        libraries.push_all(libraries);\n\
        return this;\n\
    };\n\
\n\
    this.validate = function(scenario) {\n\
        var results = $(scenario).collect(function(step) {\n\
            return _this.rank_macros(step).validate();\n\
        });\n\
        if (results.find(by_invalid_step)) throw new Error('Scenario cannot be interpreted\\n\
' + results.collect(validation_report).join('\\n\
'));\n\
    };\n\
\n\
    function by_invalid_step(result) {\n\
        return !result.valid;\n\
    }\n\
\n\
    function validation_report(result) {\n\
        return result.step + (result.valid ? '' : ' <-- ' + result.reason);\n\
    }\n\
\n\
    this.interpret = function(scenario, scenario_context, next) {\n\
        scenario_context = new Context().merge(scenario_context);\n\
        event_bus.send(EventBus.ON_SCENARIO, { scenario: scenario, ctx: scenario_context.properties });\n\
        var iterator = make_step_iterator(scenario_context, next);\n\
        $(scenario).each_async(iterator, next);\n\
    };\n\
\n\
    var make_step_iterator = function(scenario_context, next) {\n\
        var iterator = function(step, index, callback) {\n\
            _this.interpret_step(step, scenario_context, callback);\n\
        };\n\
        return next ? iterator : fn.asynchronize(null, iterator);\n\
    };\n\
\n\
    this.interpret_step = function(step, scenario_context, next) {\n\
        var context = new Context().merge(scenario_context);\n\
        event_bus.send(EventBus.ON_STEP, { step: step, ctx: context.properties });\n\
        this.rank_macros(step).clear_winner().interpret(step, context || {}, next);\n\
    };\n\
\n\
    this.rank_macros = function(step) {\n\
        return new Competition(step, compatible_macros(step));\n\
    };\n\
\n\
    var compatible_macros = function(step) {\n\
        return libraries.inject([], function(macros, library) {\n\
            return macros.concat(library.find_compatible_macros(step));\n\
        });\n\
    };\n\
};\n\
\n\
module.exports = Interpreter;\n\
));
require.register("acuminous-yadda/lib/LevenshteinDistanceScore.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
// Understands similarity of two strings\n\
var LevenshteinDistanceScore = function(s1, s2) {\n\
\n\
    this.value;\n\
    this.type = 'LevenshteinDistanceScore';\n\
    var distance_table;\n\
    var _this = this;\n\
\n\
    var initialise = function() {\n\
\n\
        /* jslint shadow: true */\n\
\n\
        var x = s1.length;\n\
        var y = s2.length;\n\
\n\
        distance_table = new Array(x + 1);\n\
\n\
        for (var i = 0; i <= x; i++) {\n\
            distance_table[i] = new Array(y + 1);\n\
        }\n\
\n\
        for (var i = 0; i <= x; i++) {\n\
            for (var j = 0; j <= y; j++) {\n\
                distance_table[i][j] = 0;\n\
            }\n\
        }\n\
\n\
        for (var i = 0; i <= x; i++) {\n\
            distance_table[i][0] = i;\n\
        }\n\
\n\
        for (var j = 0; j <= y; j++) {\n\
            distance_table[0][j] = j;\n\
        }\n\
    };\n\
\n\
    var score = function() {\n\
\n\
        /*jslint boss: true */\n\
        if (s1 == s2) return _this.value = 0;\n\
\n\
        for (var j = 0; j < s2.length; j++) {\n\
            for (var i = 0; i < s1.length; i++) {\n\
                if (s1[i] == s2[j]) {\n\
                    distance_table[i+1][j+1] = distance_table[i][j];\n\
                } else {\n\
                    var deletion = distance_table[i][j+1] + 1;\n\
                    var insertion = distance_table[i+1][j] + 1;\n\
                    var substitution = distance_table[i][j] + 1;\n\
                    distance_table[i+1][j+1] = Math.min(substitution, deletion, insertion);\n\
                }\n\
            }\n\
        }\n\
        _this.value = distance_table[s1.length][s2.length];\n\
    };\n\
\n\
    this.beats = function(other) {\n\
        return this.value < other.value;\n\
    };\n\
\n\
    this.equals = function(other) {\n\
        if (!other) return false;\n\
        return (this.type == other.type && this.value == other.value);\n\
    };\n\
\n\
    initialise();\n\
    score();\n\
};\n\
\n\
module.exports = LevenshteinDistanceScore;\n\
));
require.register("acuminous-yadda/lib/Library.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var Macro = require('./Macro');\n\
var Dictionary = require('./Dictionary');\n\
var $ = require('./Array');\n\
\n\
// Understands how to index macros\n\
var Library = function(dictionary) {\n\
\n\
    /* jslint shadow: true */\n\
    var dictionary = dictionary || new Dictionary();\n\
    var macros = $();\n\
    var _this = this;\n\
\n\
    this.define = function(signatures, fn, macro_context) {\n\
        $(signatures).each(function(signature) {\n\
            define_macro(signature, fn, macro_context);\n\
        });\n\
        return this;\n\
    };\n\
\n\
    var define_macro = function(signature, fn, macro_context) {\n\
        if (_this.get_macro(signature)) throw new Error('Duplicate macro: [' + signature + ']');\n\
        macros.push(new Macro(signature, dictionary.expand(signature), fn, macro_context));\n\
    };\n\
\n\
    this.get_macro = function(signature) {\n\
        return macros.find(function(other_macro) {\n\
            return other_macro.is_identified_by(signature);\n\
        });\n\
    };\n\
\n\
    this.find_compatible_macros = function(step) {\n\
        return macros.find_all(function(macro) {\n\
            return macro.can_interpret(step);\n\
        });\n\
    };\n\
};\n\
\n\
module.exports = Library;\n\
));
require.register("acuminous-yadda/lib/localisation/English.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var Language = require('./Language');\n\
\n\
module.exports = (function() {\n\
\n\
    var vocabulary = {\n\
        feature: '[Ff]eature',\n\
        scenario: '(?:[Ss]cenario|[Ss]cenario [Oo]utline)',\n\
        examples: '(?:[Ee]xamples|[Ww]here)',\n\
        pending: '(?:[Pp]ending|[Tt]odo)',\n\
        only: '(?:[Oo]nly)',\n\
        background: '[Bb]ackground',\n\
        given: '(?:[Gg]iven|[Ww]ith|[Aa]nd|[Bb]ut|[Ee]xcept)',\n\
        when: '(?:[Ww]hen|[Ii]f|[Aa]nd|[Bb]ut)',\n\
        then: '(?:[Tt]hen|[Ee]xpect|[Aa]nd|[Bb]ut)',\n\
        _steps: ['given', 'when', 'then']\n\
    };\n\
\n\
    return new Language('English', vocabulary);\n\
})();\n\
));
require.register("acuminous-yadda/lib/localisation/French.js", Function("exports, require, module",
"/* -*- coding: utf-8 -*-\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var Language = require('./Language');\n\
\n\
module.exports = (function() {\n\
\n\
    var vocabulary = {\n\
        feature: '(?:[Ff]onctionnalité)',\n\
        scenario: '(?:[Ss]cénario|[Pp]lan [Dd]u [Ss]cénario)',\n\
        examples: '(?:[Ee]xemples|[Ee]xemple|[Oo][uù])',\n\
        pending: '(?:[Ee]n attente|[Ee]n cours|[Tt]odo)',\n\
        only: '(?:[Ss]eulement])',\n\
        background: '(?:[Cc]ontexte)',\n\
        given: '(?:[Ss]oit|[ÉéEe]tant données|[ÉéEe]tant donnée|[ÉéEe]tant donnés|[ÉéEe]tant donné|[Aa]vec|[Ee]t|[Mm]ais|[Aa]ttendre)',\n\
        when: '(?:[Qq]uand|[Ll]orsqu\\'|[Ll]orsque|[Ss]i|[Ee]t|[Mm]ais)',\n\
        then: '(?:[Aa]lors|[Aa]ttendre|[Ee]t|[Mm]ais)',\n\
\n\
        _steps: [\n\
            'given', 'when', 'then',\n\
            'soit', 'etantdonnees', 'etantdonnee', 'etantdonne',\n\
            'quand', 'lorsque',\n\
            'alors'\n\
        ],\n\
        // Also aliasing French verbs for given-when-then for signature-lookup\n\
        get soit() { return this.given; },\n\
        get etantdonnees() { return this.given; },\n\
        get etantdonnee() { return this.given; },\n\
        get etantdonne() { return this.given; },\n\
        get quand() { return this.when; },\n\
        get lorsque() { return this.when; },\n\
        get alors() { return this.then; }\n\
    };\n\
\n\
    return new Language('French', vocabulary);\n\
})();\n\
));
require.register("acuminous-yadda/lib/localisation/German.js", Function("exports, require, module",
"/*\n\
* Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
*\n\
* Licensed under the Apache License, Version 2.0 (the \"License\");\n\
* you may not use this file except in compliance with the License.\n\
* You may obtain a copy of the License at\n\
*\n\
* http://www.apache.org/licenses/LICENSE-2.0\n\
*\n\
* Unless required by applicable law or agreed to in writing, software\n\
* distributed under the License is distributed on an \"AS IS\" BASIS,\n\
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
* See the License for the specific language governing permissions and\n\
* limitations under the License.\n\
*/\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var Language = require('./Language');\n\
\n\
module.exports = (function() {\n\
\n\
    var vocabulary = {\n\
        feature: '(?:[Ff]unktionalität|[Ff]eature|[Aa]spekt|[Uu]secase|[Aa]nwendungsfall)',\n\
        scenario: '(?:[Ss]zenario|[Ss]zenario( g|G)rundriss|[Gg]eschehnis)',\n\
        examples: '(?:[Bb]eispiele?)',\n\
        pending: '(?:[Tt]odo|[Oo]ffen)',\n\
        only: '(?:[Nn]ur|[Ee]inzig)',\n\
        background: '(?:[Gg]rundlage|[Hh]intergrund|[Ss]etup|[Vv]orausgesetzt)',\n\
        given: '(?:[Aa]ngenommen|[Gg]egeben( sei(en)?)?|[Mm]it|[Uu]nd|[Aa]ber|[Aa]ußer)',\n\
        when: '(?:[Ww]enn|[Ff]alls|[Uu]nd|[Aa]ber)',\n\
        then: '(?:[Dd]ann|[Ff]olglich|[Aa]ußer|[Uu]nd|[Aa]ber)',\n\
        _steps: ['given', 'when', 'then']\n\
    };\n\
\n\
    return new Language('German', vocabulary);\n\
})();\n\
));
require.register("acuminous-yadda/lib/localisation/index.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
module.exports = {\n\
    English: require('./English'),\n\
    French: require('./French'),\n\
    German: require('./German'),\n\
    Norwegian: require('./Norwegian'),\n\
    Pirate: require('./Pirate'),\n\
    Polish: require('./Polish'),\n\
    Spanish: require('./Spanish'),\n\
    Russian: require('./Russian'),\n\
\n\
    Language: require('./Language')\n\
};\n\
));
require.register("acuminous-yadda/lib/localisation/Language.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var Library = require('../Library');\n\
var $ = require('../Array');\n\
\n\
module.exports = function(name, vocabulary) {\n\
\n\
    var _this = this;\n\
\n\
    this.library = function(dictionary) {\n\
        return _this.localise_library(new Library(dictionary));\n\
    };\n\
\n\
    this.localise_library = function(library) {\n\
        $(vocabulary._steps).each(function(keyword) {\n\
            library[keyword] = function(signatures, fn, ctx) {\n\
                return $(signatures).each(function(signature) {\n\
                    signature = prefix_signature(_this.localise(keyword), signature);\n\
                    return library.define(signature, fn, ctx);\n\
                });\n\
            };\n\
        });\n\
        return library;\n\
    };\n\
\n\
    var prefix_signature = function(prefix, signature) {\n\
        var regex_delimiters = new RegExp('^/|/$', 'g');\n\
        var start_of_signature = new RegExp(/^(?:\\^)?/);\n\
        var one_or_more_spaces = '\\\\s+';\n\
        return signature.toString().replace(regex_delimiters, '').replace(start_of_signature, prefix + one_or_more_spaces);\n\
    };\n\
\n\
    this.localise = function(keyword) {\n\
        if (vocabulary[keyword] === undefined) throw new Error('Keyword \"' + keyword + '\" has not been translated into ' + name + '.');\n\
        return vocabulary[keyword];\n\
    };\n\
};\n\
));
require.register("acuminous-yadda/lib/localisation/Pirate.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var Language = require('./Language');\n\
\n\
module.exports = (function() {\n\
\n\
    var vocabulary = {\n\
        feature: '(?:[Tt]ale|[Yy]arn)',\n\
        scenario: '(?:[Aa]dventure|[Ss]ortie)',\n\
        examples: '[Ww]herest',\n\
        pending: '[Bb]rig',\n\
        only: '[Bb]lack [Ss]pot',\n\
        background: '[Aa]ftground',\n\
        given: '(?:[Gg]iveth|[Ww]ith|[Aa]nd|[Bb]ut|[Ee]xcept)',\n\
        when: '(?:[Ww]hence|[Ii]f|[Aa]nd|[Bb]ut)',\n\
        then: '(?:[Tt]hence|[Ee]xpect|[Aa]nd|[Bb]ut)',\n\
        _steps: ['given', 'when', 'then', 'giveth', 'whence', 'thence'],\n\
        // Also aliasing Pirate verbs for given-when-then for signature-lookup\n\
        get giveth() { return this.given; },\n\
        get whence() { return this.when; },\n\
        get thence() { return this.then; }\n\
\n\
    };\n\
\n\
    return new Language('Pirate', vocabulary);\n\
})();\n\
));
require.register("acuminous-yadda/lib/localisation/Polish.js", Function("exports, require, module",
"/* jslint node: true */\n\
\"use strict\";\n\
\n\
var Language = require('./Language');\n\
\n\
module.exports = (function() {\n\
\n\
    var vocabulary = {\n\
        feature: '(?:[Ww]łaściwość|[Ff]unkcja|[Aa]spekt|[Pp]otrzeba [Bb]iznesowa)',\n\
        scenario: '(?:[Ss]cenariusz|[Ss]zablon [Ss]cenariusza)',\n\
        examples: '[Pp]rzykłady',\n\
        pending: '(?:[Oo]czekujący|[Nn]iezweryfikowany|[Tt]odo)',\n\
        only: '[Tt]ylko',\n\
        background: '[Zz]ałożenia',\n\
        given: '(?:[Zz]akładając|[Mm]ając|[Oo]raz|[Ii]|[Aa]le)',\n\
        when: '(?:[Jj]eżeli|[Jj]eśli|[Gg]dy|[Kk]iedy|[Oo]raz|[Ii]|[Aa]le)',\n\
        then: '(?:[Ww]tedy|[Oo]raz|[Ii]|[Aa]le)',\n\
        _steps: [\n\
            'given', 'when', 'then',\n\
            'zakladajac', 'majac',\n\
            'jezeli', 'jesli', 'gdy', 'kiedy',\n\
            'wtedy'\n\
        ],\n\
        // Also aliasing Polish verbs for given-when-then for signature-lookup\n\
        get zakladajac() { return this.given; },\n\
        get majac() { return this.given; },\n\
        get jezeli() { return this.when; },\n\
        get jesli() { return this.when; },\n\
        get gdy() { return this.when; },\n\
        get kiedy() { return this.when; },\n\
        get wtedy() { return this.then; }\n\
    };\n\
\n\
    return new Language('Polish', vocabulary);\n\
})();\n\
));
require.register("acuminous-yadda/lib/localisation/Russian.js", Function("exports, require, module",
"/* -*- coding: utf-8 -*-\n\
 * Author: Marat Dyatko\n\
 * https://github.com/vectart\n\
 *\n\
 * Inspired by Gherkin vocabulary\n\
 * https://github.com/cucumber/gherkin/blob/master/lib/gherkin/i18n.json\n\
 *\n\
 * Also considered syntax highlight of Cucumber Sublime bundle\n\
 * https://github.com/drewda/cucumber-sublime-bundle/blob/master/Cucumber%20Plain%20Text%20Feature.tmLanguage\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var Language = require('./Language');\n\
\n\
module.exports = (function() {\n\
\n\
    var vocabulary = {\n\
        feature: '(?:[Фф]ункция|[Фф]ункционал|[Сс]войство)',\n\
        scenario: 'Сценарий',\n\
        examples: 'Примеры?',\n\
        pending: '(?:[Ww]ip|[Tt]odo)',\n\
        only: 'Только',\n\
        background: '(?:[Пп]редыстория|[Кк]онтекст)',\n\
        given: '(?:[Дд]опустим|[Дд]ано|[Пп]усть|[Ии]|[Н]о)(?:\\\\s[Яя])?',\n\
        when: '(?:[Ее]сли|[Кк]огда|[Ии]|[Н]о)(?:\\\\s[Яя])?',\n\
        then: '(?:[Тт]о|[Тт]огда|[Ии]|[Н]о)(?:\\\\s[Яя])?',\n\
        _steps: ['given', 'when', 'then']\n\
    };\n\
\n\
    return new Language('Russian', vocabulary);\n\
})();\n\
));
require.register("acuminous-yadda/lib/localisation/Spanish.js", Function("exports, require, module",
"/* -*- coding: utf-8 -*-\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var Language = require('./Language');\n\
\n\
module.exports = (function() {\n\
\n\
    var vocabulary = {\n\
        feature: '(?:[Ff]uncionalidad|[Cc]aracterística)',\n\
        scenario: '(?:[Ee]scenario|[Cc]aso)',\n\
        examples: '(?:[Ee]jemplos|[Ee]jemplo)',\n\
        pending: '[Pp]endiente',\n\
        only: '[S]ólo',\n\
        background: '[Ff]ondo',\n\
        given: '(?:[Ss]ea|[Ss]ean|[Dd]ado|[Dd]ada|[Dd]ados|[Dd]adas)',\n\
        when: '(?:[Cc]uando|[Ss]i|[Qq]ue)',\n\
        then: '(?:[Ee]ntonces)',\n\
\n\
        _steps: [\n\
            'given', 'when', 'then',\n\
            'sea', 'sean', 'dado', 'dada','dados', 'dadas',\n\
            'cuando', 'si',\n\
            'entonces'\n\
        ],\n\
\n\
        get sea() { return this.given; },\n\
        get sean() { return this.given; },\n\
        get dado() { return this.given; },\n\
        get dada() { return this.given; },\n\
        get dados() { return this.given; },\n\
        get dadas() { return this.given; },\n\
        get cuando() { return this.when; },\n\
        get si() { return this.when; },\n\
        get entonces() { return this.then; }\n\
    };\n\
\n\
    return new Language('Spanish', vocabulary);\n\
})();\n\
));
require.register("acuminous-yadda/lib/localisation/Norwegian.js", Function("exports, require, module",
"﻿/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var Language = require('./Language');\n\
\n\
module.exports = (function() {\n\
\n\
    var vocabulary = {\n\
        feature: '[Ee]genskap',\n\
        scenario: '[Ss]cenario',\n\
        examples: '[Ee]ksempler',\n\
        pending: '[Aa]vventer',\n\
        only: '[Bb]are',\n\
        background: '[Bb]akgrunn',\n\
        given: '(?:[Gg]itt|[Mm]ed|[Oo]g|[Mm]en|[Uu]nntatt)',\n\
        when: '(?:[Nn]år|[Oo]g|[Mm]en)',\n\
        then: '(?:[Ss]å|[Ff]forvent|[Oo]g|[Mm]en)',\n\
        _steps: ['given', 'when', 'then', 'gitt', 'når', 'så'],\n\
        // Also aliasing Norwegian verbs for given-when-then for signature-lookup\n\
        get gitt() { return this.given; },\n\
        get når() { return this.when; },\n\
        get så() { return this.then; }\n\
    };\n\
\n\
    return new Language('Norwegian', vocabulary);\n\
})();\n\
));
require.register("acuminous-yadda/lib/Macro.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var fn = require('./fn');\n\
var Context = require('./Context');\n\
var RegularExpression = require('./RegularExpression');\n\
var EventBus = require('./EventBus');\n\
\n\
// Understands how to invoke a step\n\
var Macro = function(signature, signature_pattern, macro, macro_context) {\n\
\n\
    /* jslint shadow: true */\n\
    var signature_pattern = new RegularExpression(signature_pattern);\n\
    var macro = macro || fn.async_noop;\n\
    var event_bus = EventBus.instance();\n\
    var _this = this;\n\
\n\
    var init = function(signature, signature_pattern) {\n\
        _this.signature = normalise(signature);\n\
    };\n\
\n\
    this.is_identified_by = function(other_signature) {\n\
        return this.signature == normalise(other_signature);\n\
    };\n\
\n\
    this.can_interpret = function(step) {\n\
        return signature_pattern.test(step);\n\
    };\n\
\n\
    this.interpret = function(step, scenario_context, next) {\n\
        var context = new Context().merge(macro_context).merge(scenario_context);\n\
        var args = signature_pattern.groups(step);\n\
        event_bus.send(EventBus.ON_EXECUTE, { step: step, ctx: context.properties, pattern: signature_pattern.toString(), args: args });\n\
        fn.invoke(macro, context.properties, args.concat(next));\n\
    };\n\
\n\
    this.levenshtein_signature = function() {\n\
        return signature_pattern.without_expressions();\n\
    };\n\
\n\
    var normalise = function(signature) {\n\
        return new RegExp(signature).toString();\n\
    };\n\
\n\
    this.toString = function() {\n\
        return this.signature;\n\
    };\n\
\n\
    init(signature, signature_pattern);\n\
};\n\
\n\
module.exports = Macro;\n\
));
require.register("acuminous-yadda/lib/parsers/FeatureParser.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
 /* jslint node: true */\n\
 \"use strict\";\n\
\n\
var $ = require('../Array');\n\
var fn = require('../fn');\n\
\n\
var English = require('../localisation/English');\n\
\n\
var FeatureParser = function(language) {\n\
\n\
    /* jslint shadow: true */\n\
    var language = language || English;\n\
\n\
    var FEATURE_REGEX = new RegExp('^\\\\s*' + language.localise('feature') + ':\\\\s*(.*)', 'i');\n\
    var SCENARIO_REGEX = new RegExp('^\\\\s*' + language.localise('scenario') + ':\\\\s*(.*)', 'i');\n\
    var BACKGROUND_REGEX = new RegExp('^\\\\s*' + language.localise('background') + ':\\\\s*(.*)', 'i');\n\
    var EXAMPLES_REGEX = new RegExp('^\\\\s*' + language.localise('examples') + ':', 'i');\n\
    var TEXT_REGEX = new RegExp('^\\\\s*([^\\\\s].*)', 'i');\n\
    var SINGLE_LINE_COMMENT_REGEX = new RegExp('^\\\\s*#');\n\
    var MULTI_LINE_COMMENT_REGEX = new RegExp('^\\\\s*#{3,}');\n\
    var BLANK_REGEX = new RegExp('^\\\\s*$');\n\
    var SIMPLE_ANNOTATION_REGEX = new RegExp('^\\\\s*@([^=]*)$');\n\
    var NVP_ANNOTATION_REGEX = new RegExp('^\\\\s*@([^=]*)=(.*)$');\n\
\n\
    var specification;\n\
    var comment;\n\
    var line;\n\
    var line_number = 0;\n\
\n\
    this.parse = function(text, next) {\n\
        reset();\n\
        split(text).each(parse_line);\n\
        return next && next(specification.export()) || specification.export();\n\
    };\n\
\n\
    function reset() {\n\
        specification = new Specification();\n\
        comment = false;\n\
        line_number = 0;\n\
    }\n\
\n\
    function split(text) {\n\
        return $(text.split(/\\r\\n\
|\\n\
/));\n\
    }\n\
\n\
    function parse_line(line, index) {\n\
        /* jslint boss: true */\n\
        var match;\n\
        try {\n\
            if (match = MULTI_LINE_COMMENT_REGEX.test(line)) return comment = !comment;\n\
            if (comment) return;\n\
            if (match = SINGLE_LINE_COMMENT_REGEX.test(line)) return;\n\
            if (match = SIMPLE_ANNOTATION_REGEX.exec(line)) return specification.handle('Annotation', { key: match[1], value: true });\n\
            if (match = NVP_ANNOTATION_REGEX.exec(line)) return specification.handle('Annotation', { key: match[1], value: match[2] });\n\
            if (match = FEATURE_REGEX.exec(line)) return specification.handle('Feature', match[1]);\n\
            if (match = SCENARIO_REGEX.exec(line)) return specification.handle('Scenario', match[1]);\n\
            if (match = BACKGROUND_REGEX.exec(line)) return specification.handle('Background', match[1]);\n\
            if (match = EXAMPLES_REGEX.exec(line)) return specification.handle('Examples');\n\
            if (match = BLANK_REGEX.test(line)) return specification.handle('Blank');\n\
            if (match = TEXT_REGEX.exec(line)) return specification.handle('Text', match[1]);\n\
        } catch (e) {\n\
            throw new Error('Error parsing line ' + (index  + 1) + ', \"' + line + '\".\\n\
' + e.message);\n\
        }\n\
    }\n\
};\n\
\n\
var Handlers = function(handlers) {\n\
\n\
    /* jslint shadow: true */\n\
    var handlers = handlers || {};\n\
\n\
    this.register = function(event, handler) {\n\
        handlers[event] = handler;\n\
    };\n\
\n\
    this.unregister = function(event) {\n\
        delete handlers[event];\n\
    };\n\
\n\
    this.find = function(event) {\n\
        if (!handlers[event.toLowerCase()]) throw new Error(event + ' is unexpected at this time');\n\
        return { handle: handlers[event.toLowerCase()] };\n\
    };\n\
};\n\
\n\
var Specification = function() {\n\
\n\
    var current_element = this;\n\
    var feature;\n\
    var annotations = {};\n\
    var handlers = new Handlers({\n\
        text: fn.noop,\n\
        blank: fn.noop,\n\
        annotation: stash_annotation,\n\
        feature: start_feature,\n\
        scenario: start_scenario,\n\
        background: start_background,\n\
    });\n\
\n\
    function stash_annotation(event, annotation) {\n\
        handlers.unregister('background');\n\
        annotations[annotation.key] = annotation.value;\n\
        annotations[annotation.key.toLowerCase().replace(/\\W/g, '_')] = annotation.value;\n\
    }\n\
\n\
    function start_feature(event, title) {\n\
        /* jslint boss: true */\n\
        return feature = new Feature(title, annotations, {});\n\
    }\n\
\n\
    function start_scenario(event, title) {\n\
        feature = new Feature(title, {}, annotations);\n\
        return feature.on(event, title);\n\
    }\n\
\n\
    var start_background = start_scenario;\n\
\n\
    this.handle = function(event, data) {\n\
        current_element = current_element.on(event, data);\n\
    };\n\
\n\
    this.on = function(event, data) {\n\
        return handlers.find(event).handle(event, data) || this;\n\
    };\n\
\n\
    this.export = function() {\n\
        if (!feature) throw new Error('A feature must contain one or more scenarios');\n\
        return feature.export();\n\
    };\n\
};\n\
\n\
var Feature = function(title, annotations, stashed_annotations) {\n\
\n\
    var description = [];\n\
    var scenarios = [];\n\
    var background = new NullBackground();\n\
    var handlers = new Handlers({\n\
        text: capture_description,\n\
        blank: end_description,\n\
        annotation: stash_annotation,\n\
        scenario: start_scenario,\n\
        background: start_background\n\
    });\n\
    var _this = this;\n\
\n\
    function start_background(event, title) {\n\
        background = new Background(title, _this);\n\
        stashed_annotations = {};\n\
        return background;\n\
    }\n\
\n\
    function stash_annotation(event, annotation) {\n\
        handlers.unregister('background');\n\
        stashed_annotations[annotation.key] = annotation.value;\n\
        stashed_annotations[annotation.key.toLowerCase().replace(/\\W/g, '_')] = annotation.value;\n\
    }\n\
\n\
    function capture_description(event, text) {\n\
        description.push(text);\n\
    }\n\
\n\
    function end_description() {\n\
        handlers.unregister('text');\n\
        handlers.register('blank', fn.noop);\n\
    }\n\
\n\
    function start_scenario(event, title) {\n\
        var scenario = new Scenario(title, background, stashed_annotations, _this);\n\
        scenarios.push(scenario);\n\
        stashed_annotations = {};\n\
        return scenario;\n\
    }\n\
\n\
    function validate() {\n\
        if (scenarios.length === 0) throw new Error('Feature requires one or more scenarios');\n\
    }\n\
\n\
    this.on = function(event, data) {\n\
        return handlers.find(event).handle(event, data) || this;\n\
    };\n\
\n\
    this.export = function() {\n\
        validate();\n\
        return {\n\
            title: title,\n\
            annotations: annotations,\n\
            description: description,\n\
            scenarios: $(scenarios).collect(function(scenario) {\n\
                return scenario.export();\n\
            }).flatten().naked()\n\
        };\n\
    };\n\
};\n\
\n\
var Background = function(title, feature) {\n\
\n\
    var steps = [];\n\
    var handlers = new Handlers({\n\
        text: fn.noop,\n\
        blank: end_description,\n\
        annotation: stash_annotation,\n\
        scenario: start_scenario\n\
    });\n\
    var _this = this;\n\
\n\
    function end_description() {\n\
        handlers.register('text', capture_step);\n\
        handlers.register('blank', fn.noop);\n\
    }\n\
\n\
    function capture_step(event, text) {\n\
        steps.push(text);\n\
    }\n\
\n\
    function stash_annotation(event, annotation) {\n\
        validate();\n\
        return feature.on(event, annotation);\n\
    }\n\
\n\
    function start_scenario(event, data) {\n\
        validate();\n\
        return feature.on(event, data);\n\
    }\n\
\n\
    function validate() {\n\
        if (steps.length === 0) throw new Error('Background requires one or more steps');\n\
    }\n\
\n\
    this.on = function(event, data) {\n\
        return handlers.find(event).handle(event, data) || this;\n\
    };\n\
\n\
    this.export = function() {\n\
        validate();\n\
        return {\n\
            steps: steps\n\
        };\n\
    };\n\
};\n\
\n\
var NullBackground = function() {\n\
    var handlers = new Handlers();\n\
\n\
    this.on = function(event, data) {\n\
        return handlers.find(event).handle(event, data) || this;\n\
    };\n\
\n\
    this.export = function() {\n\
        return {\n\
            steps: []\n\
        };\n\
    };\n\
};\n\
\n\
var Scenario = function(title, background, annotations, feature) {\n\
\n\
    var description = [];\n\
    var steps = [];\n\
    var examples;\n\
    var handlers = new Handlers({\n\
        text: capture_step,\n\
        blank: fn.noop,\n\
        annotation: start_scenario,\n\
        scenario: start_scenario,\n\
        examples: start_examples\n\
    });\n\
    var _this = this;\n\
\n\
    function capture_step(event, text) {\n\
        steps.push(text);\n\
    }\n\
\n\
    function start_scenario(event, data) {\n\
        validate();\n\
        return feature.on(event, data);\n\
    }\n\
\n\
    function start_examples(event, data) {\n\
        validate();\n\
        /* jslint boss: true */\n\
        return examples = new Examples(_this);\n\
    }\n\
\n\
    function validate() {\n\
        if (steps.length === 0) throw new Error('Scenario requires one or more steps');\n\
    }\n\
\n\
    this.on = function(event, data) {\n\
        return handlers.find(event).handle(event, data) || this;\n\
    };\n\
\n\
    this.export = function() {\n\
        validate();\n\
        var result = {\n\
            title: title,\n\
            annotations: annotations,\n\
            description: description,\n\
            steps: background.export().steps.concat(steps)\n\
        };\n\
        return examples ? examples.expand(result) : result;\n\
    };\n\
};\n\
\n\
var Examples = function(scenario) {\n\
\n\
    var SURROUNDING_WHITESPACE_REGEX = /^\\s+|\\s+$/g;\n\
\n\
    var headings = [];\n\
    var examples = $();\n\
    var handlers = new Handlers({\n\
        text: capture_headings,\n\
        blank: fn.noop,\n\
        annotation: start_scenario,\n\
        scenario: start_scenario,\n\
    });\n\
    var _this = this;\n\
\n\
    function capture_headings(event, data) {\n\
        handlers.register('text', capture_example);\n\
        headings = split(data).collect(function(column) {\n\
            return column.replace(SURROUNDING_WHITESPACE_REGEX, '');\n\
        }).naked();\n\
    }\n\
\n\
    function capture_example(event, data) {\n\
        var fields = split(data, headings.length);\n\
        var example = {};\n\
        fields.each(function(field, index) {\n\
            example[headings[index]] = field.replace(SURROUNDING_WHITESPACE_REGEX, '');\n\
        });\n\
        examples.push(example);\n\
    }\n\
\n\
    function split(row, number_of_fields) {\n\
        var fields = $(row.split('|'));\n\
        if (number_of_fields !== undefined && number_of_fields != fields.length) {\n\
            throw new Error('Incorrect number of fields in example table. Expected ' + number_of_fields + ' but found ' + fields.length);\n\
        }\n\
        return fields;\n\
    }\n\
\n\
    function start_scenario(event, data) {\n\
        validate();\n\
        return scenario.on(event, data);\n\
    }\n\
\n\
    function validate() {\n\
        if (headings.length === 0) throw new Error('Examples table requires one or more headings');\n\
        if (examples.length === 0) throw new Error('Examples table requires one or more rows');\n\
    }\n\
\n\
    this.on = function(event, data) {\n\
        return handlers.find(event).handle(event, data) || this;\n\
    };\n\
\n\
    this.expand = function(scenario) {\n\
        validate();\n\
        return examples.collect(function(example) {\n\
            return {\n\
                title: substitute(example, scenario.title),\n\
                annotations: shallowClone(scenario.annotations),\n\
                description: substitute_all(example, scenario.description),\n\
                steps: substitute_all(example, scenario.steps)\n\
            };\n\
        }).naked();\n\
    };\n\
\n\
    function shallowClone(source) {\n\
        var dest = {};\n\
        for (var key in source) {\n\
            dest[key] = source[key];\n\
        }\n\
        return dest;\n\
    }\n\
\n\
    function substitute_all(example, lines) {\n\
        return $(lines).collect(function(line) {\n\
            return substitute(example, line);\n\
        }).naked();\n\
    }\n\
\n\
    function substitute(example, line) {\n\
        for (var heading in example) {\n\
            line = line.replace(new RegExp('\\\\[\\\\s*' + heading + '\\\\s*\\\\]', 'g'), example[heading]);\n\
        }\n\
        return line;\n\
    }\n\
};\n\
\n\
module.exports = FeatureParser;\n\
));
require.register("acuminous-yadda/lib/parsers/FeatureFileParser.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
 /* jslint node: true */\n\
 \"use strict\";\n\
\n\
var FeatureFileParser = function(language) {\n\
\n\
    // Requiring fs locally so it doesn't break component\n\
    var fs = require('fs');\n\
    var FeatureParser = require('./FeatureParser');\n\
    var parser = new FeatureParser(language);\n\
\n\
    this.parse = function(file, next) {\n\
        var text = fs.readFileSync(file, 'utf8');\n\
        var feature = parser.parse(text);\n\
        return next && next(feature) || feature;\n\
    };\n\
};\n\
\n\
module.exports = FeatureFileParser;\n\
));
require.register("acuminous-yadda/lib/parsers/StepParser.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var $ = require('../Array');\n\
\n\
var StepParser = function() {\n\
\n\
    var NON_BLANK_REGEX = /[^\\s]/;\n\
\n\
    this.parse = function(text, next) {\n\
        var steps = split(text).find_all(non_blanks);\n\
        return next && next(steps) || steps;\n\
    };\n\
\n\
    var split = function(text) {\n\
        return $(text.split(/\\n\
/));\n\
    };\n\
\n\
    var non_blanks = function(text) {\n\
        return text && NON_BLANK_REGEX.test(text);\n\
    };\n\
};\n\
\n\
module.exports = StepParser;\n\
));
require.register("acuminous-yadda/lib/parsers/index.js", Function("exports, require, module",
"/* jslint node: true */\n\
\"use strict\";\n\
\n\
module.exports = {\n\
    StepParser: require('./StepParser'),\n\
    FeatureParser: require('./FeatureParser'),\n\
    FeatureFileParser: require('./FeatureFileParser')\n\
};\n\
));
require.register("acuminous-yadda/lib/Platform.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
/* jslint browser: true */\n\
/* jslint phantom: true */\n\
\"use strict\";\n\
\n\
module.exports = Platform;\n\
\n\
function Platform() {\n\
\n\
    function get_container() {\n\
        if (is_browser()) return window;\n\
        if (is_phantom()) return phantom;\n\
        if (is_node()) return global;\n\
    }\n\
\n\
    function is_node() {\n\
        return typeof process != 'undefined' &&\n\
               typeof GLOBAL != 'undefined' &&\n\
               typeof __dirname != 'undefined';\n\
    }\n\
\n\
    function is_browser() {\n\
        return typeof window != 'undefined';\n\
    }\n\
\n\
    function is_phantom() {\n\
        return typeof phantom != 'undefined';\n\
    }\n\
\n\
    return {\n\
        get_container: get_container,\n\
        is_node: is_node,\n\
        is_browser: is_browser,\n\
        is_phantom: is_phantom\n\
    };\n\
\n\
}\n\
));
require.register("acuminous-yadda/lib/plugins/CasperPlugin.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
if (!module.client) {\n\
    var fs = require('fs');\n\
    global.process = global.process || {\n\
        cwd: function() {\n\
            return fs.workingDirectory;\n\
        }\n\
    };\n\
}\n\
\n\
\n\
module.exports = function(yadda, casper) {\n\
\n\
    var EventBus = require('yadda').EventBus;\n\
\n\
    yadda.interpreter.interpret_step = function(step, ctx, next) {\n\
\n\
        var _this = this;\n\
        casper.then(function() {\n\
            casper.test.info(step);\n\
            EventBus.instance().send(EventBus.ON_STEP, { step: step, ctx: ctx });\n\
            _this.rank_macros(step).clear_winner().interpret(step, ctx, next);\n\
        });\n\
    };\n\
\n\
    casper.yadda = function(script, ctx) {\n\
        if (script === undefined) return this;\n\
        yadda.run(script, ctx);\n\
    };\n\
};\n\
));
require.register("acuminous-yadda/lib/plugins/index.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
module.exports = {\n\
    casper: require('./CasperPlugin'),\n\
    get mocha() {\n\
        var legacyPlugin = require('./MochaPlugin');\n\
        legacyPlugin.AsyncScenarioLevelPlugin = require('./mocha/AsyncScenarioLevelPlugin');\n\
        legacyPlugin.SyncScenarioLevelPlugin = require('./mocha/SyncScenarioLevelPlugin');\n\
        legacyPlugin.AsyncStepLevelPlugin = require('./mocha/AsyncStepLevelPlugin');\n\
        legacyPlugin.SyncStepLevelPlugin = require('./mocha/SyncStepLevelPlugin');\n\
        legacyPlugin.ScenarioLevelPlugin = require('./mocha/ScenarioLevelPlugin');\n\
        legacyPlugin.StepLevelPlugin = require('./mocha/StepLevelPlugin');\n\
        return legacyPlugin;\n\
    },\n\
    get jasmine() {\n\
        return this.mocha;\n\
    }\n\
};\n\
));
require.register("acuminous-yadda/lib/plugins/MochaPlugin.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
 \n\
/* jslint node: true */\n\
/* jslint browser: true */\n\
/* global describe, xdescribe, it, xit */\n\
\"use strict\";\n\
\n\
if (!module.client) {\n\
    var fs = require('fs');\n\
}\n\
var English = require('../localisation/English');\n\
var FeatureParser = require('../parsers/FeatureParser');\n\
var $ = require('../Array');\n\
\n\
module.exports = function(options) {\n\
\n\
    /* jslint shadow: true */\n\
    var options = options || {};\n\
    var language = options.language || English;\n\
    var parser = options.parser || new FeatureParser(language);\n\
    var mode = options.mode || 'async';\n\
    var feature;\n\
\n\
    if (options.deprecation_warning !== false) {\n\
        console.log('The MochaPlugin is deprecated as of 0.10.0 and will be removed in 0.12.0');\n\
        console.log('Replace it with one of AsyncScenarioLevelPlugin, SyncScenarioLevelPlugin, AsyncStepLevelPlugin or SyncStepLevelPlugin');\n\
        console.log('To disable this message use Yadda.plugins.mocha({deprecation_warning: false})');\n\
        console.log('See the readme for more details');\n\
    }\n\
\n\
    if (module.client) {\n\
        feature = function (text, next) {\n\
            parser.parse(text, function(feature) {\n\
                var _describe = feature.annotations[language.localise('pending')] ? xdescribe : describe;\n\
                _describe(feature.title, function() {\n\
                    next(feature);\n\
                });\n\
            });\n\
        };\n\
    } else {\n\
        feature = function (filenames, next) {\n\
            $(filenames).each(function(filename) {\n\
                var text = fs.readFileSync(filename, 'utf8');\n\
                parser.parse(text, function(feature) {\n\
                    var _describe = feature.annotations[language.localise('pending')] ? xdescribe : describe;\n\
                    _describe(feature.title || filename, function() {\n\
                        next(feature);\n\
                    });\n\
                });\n\
            });\n\
        };\n\
    }\n\
\n\
    function async_scenarios(scenarios, next) {\n\
        $(scenarios).each(function(scenario) {\n\
            var _it = scenario.annotations[language.localise('pending')] ? xit : it;\n\
            _it(scenario.title, function(done) {\n\
                next(scenario, done);\n\
            });\n\
        });\n\
    }\n\
\n\
    function sync_scenarios(scenarios, next) {\n\
        $(scenarios).each(function(scenario) {\n\
            var _it = scenario.annotations[language.localise('pending')] ? xit : it;\n\
            _it(scenario.title, function() {\n\
                next(scenario);\n\
            });\n\
        });\n\
    }\n\
\n\
    if (typeof GLOBAL !== 'undefined') {\n\
        GLOBAL.features = GLOBAL.feature = feature;\n\
        GLOBAL.scenarios = mode == 'async' ? async_scenarios : sync_scenarios;\n\
    }\n\
\n\
    if (typeof window !== 'undefined') {\n\
        window.features = window.feature = feature;\n\
        window.scenarios = mode == 'async' ? async_scenarios : sync_scenarios;\n\
    }\n\
};\n\
));
require.register("acuminous-yadda/lib/plugins/mocha/AsyncScenarioLevelPlugin.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var $ = require('../../Array');\n\
var Platform = require('../../Platform');\n\
var BasePlugin = require('./BasePlugin');\n\
\n\
module.exports.init = function(options) {\n\
\n\
    /* jslint shadow: true */\n\
    var options = options || {};\n\
    var platform = new Platform();\n\
    var container = options.container || platform.get_container();\n\
\n\
    var base_plugin = BasePlugin.create(options);\n\
\n\
    function scenarios(scenarios, iterator) {\n\
        if (!options.silenceDeprecations) {\n\
            console.log('*******************************************************************************');\n\
            console.log('* AsyncScenarioLevelPlugin has been deprecated and will soon be removed.      *');\n\
            console.log('* Use the ScenarioLevelPlugin instead.                                        *');\n\
            console.log('* To turn off this message add silenceDeprecations: true to the init options. *');\n\
            console.log('*******************************************************************************');\n\
        }\n\
        $(scenarios).each(function(scenario) {\n\
            base_plugin.it_async(scenario.title, scenario, iterator);\n\
        });\n\
    }\n\
\n\
    container.featureFiles = container.featureFile = base_plugin.featureFiles;\n\
    container.features = container.feature = base_plugin.features;\n\
    container.scenarios = container.scenario = scenarios;\n\
};\n\
));
require.register("acuminous-yadda/lib/plugins/mocha/AsyncStepLevelPlugin.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var $ = require('../../Array');\n\
var Platform = require('../../Platform');\n\
var BasePlugin = require('./BasePlugin');\n\
\n\
module.exports.init = function(options) {\n\
\n\
    /* jslint shadow: true */\n\
    var options = options || {};\n\
    var platform = new Platform();\n\
    var container = options.container || platform.get_container();\n\
\n\
    var base_plugin = BasePlugin.create(options);\n\
\n\
    function scenarios(scenarios, iterator) {\n\
        if (!options.silenceDeprecations) {\n\
            console.log('*******************************************************************************');\n\
            console.log('* AsyncStepLevelPlugin has been deprecated and will soon be removed.          *');\n\
            console.log('* Use the ScenarioLevelPlugin instead.                                        *');\n\
            console.log('* To turn off this message add silenceDeprecations: true to the init options. *');\n\
            console.log('*******************************************************************************');\n\
        }\n\
        $(scenarios).each(function(scenario) {\n\
            base_plugin.describe(scenario.title, scenario, iterator);\n\
        });\n\
    }\n\
\n\
    function steps(steps, iterator) {\n\
        var abort;\n\
        $(steps).each(function(step) {\n\
            base_plugin.it_async(step, step, function(step, done) {\n\
                if (abort) return done();\n\
                iterator(step, function(err) {\n\
                    if (err) abort = true;\n\
                    done(err);\n\
                });\n\
            });\n\
        });\n\
    }\n\
\n\
    container.featureFiles = container.featureFile = base_plugin.featureFiles;\n\
    container.features = container.feature = base_plugin.features;\n\
    container.scenarios = container.scenario = scenarios;\n\
    container.steps = steps;\n\
};\n\
));
require.register("acuminous-yadda/lib/plugins/mocha/SyncScenarioLevelPlugin.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var $ = require('../../Array');\n\
var Platform = require('../../Platform');\n\
var BasePlugin = require('./BasePlugin');\n\
\n\
module.exports.init = function(options) {\n\
\n\
    /* jslint shadow: true */\n\
    var options = options || {};\n\
    var platform = new Platform();\n\
    var container = options.container || platform.get_container();\n\
\n\
    var base_plugin = BasePlugin.create(options);\n\
\n\
    function scenarios(scenarios, iterator) {\n\
        if (!options.silenceDeprecations) {\n\
            console.log('*******************************************************************************');\n\
            console.log('* SyncScenarioLevelPlugin has been deprecated and will soon be removed.       *');\n\
            console.log('* Use the ScenarioLevelPlugin instead.                                        *');\n\
            console.log('* To turn off this message add silenceDeprecations: true to the init options. *');\n\
            console.log('*******************************************************************************');\n\
        }\n\
        $(scenarios).each(function(scenario) {\n\
            base_plugin.it_sync(scenario.title, scenario, iterator);\n\
        });\n\
    }\n\
\n\
    container.featureFiles = container.featureFile = base_plugin.featureFiles;\n\
    container.features = container.feature = base_plugin.features;\n\
    container.scenarios = container.scenario = scenarios;\n\
};\n\
));
require.register("acuminous-yadda/lib/plugins/mocha/SyncStepLevelPlugin.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var $ = require('../../Array');\n\
var Platform = require('../../Platform');\n\
var BasePlugin = require('./BasePlugin');\n\
\n\
module.exports.init = function(options) {\n\
\n\
    /* jslint shadow: true */\n\
    var options = options || {};\n\
    var platform = new Platform();\n\
    var container = options.container || platform.get_container();\n\
\n\
    var base_plugin = BasePlugin.create(options);\n\
\n\
    function scenarios(scenarios, iterator) {\n\
        if (!options.silenceDeprecations) {\n\
            console.log('*******************************************************************************');\n\
            console.log('* SyncStepLevelPlugin has been deprecated and will soon be removed.           *');\n\
            console.log('* Use the ScenarioLevelPlugin instead.                                        *');\n\
            console.log('* To turn off this message add silenceDeprecations: true to the init options. *');\n\
            console.log('*******************************************************************************');\n\
        }\n\
        $(scenarios).each(function(scenario) {\n\
            base_plugin.describe(scenario.title, scenario, iterator);\n\
        });\n\
    }\n\
\n\
    function steps(steps, iterator) {\n\
        var abort;\n\
        $(steps).each(function(step) {\n\
            base_plugin.it_sync(step, step, function(step) {\n\
                if (abort) return;\n\
                abort = true;\n\
                iterator(step);\n\
                abort = false;\n\
            });\n\
        });\n\
    }\n\
\n\
\n\
    container.featureFiles = container.featureFile = base_plugin.featureFiles;\n\
    container.features = container.feature = base_plugin.features;\n\
    container.scenarios = container.scenario = scenarios;\n\
    container.steps = steps;\n\
};\n\
));
require.register("acuminous-yadda/lib/plugins/mocha/ScenarioLevelPlugin.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var $ = require('../../Array');\n\
var Platform = require('../../Platform');\n\
var BasePlugin = require('./BasePlugin');\n\
\n\
module.exports.init = function(options) {\n\
\n\
    /* jslint shadow: true */\n\
    var options = options || {};\n\
    var platform = new Platform();\n\
    var container = options.container || platform.get_container();\n\
\n\
    var base_plugin = BasePlugin.create(options);\n\
\n\
    function scenarios(scenarios, iterator) {\n\
        $(scenarios).each(function(scenario) {\n\
            var itFn = iterator.length == 1 ? base_plugin.it_sync : base_plugin.it_async;\n\
            itFn(scenario.title, scenario, iterator);\n\
        });\n\
    }\n\
\n\
    container.featureFiles = container.featureFile = base_plugin.featureFiles;\n\
    container.features = container.feature = base_plugin.features;\n\
    container.scenarios = container.scenario = scenarios;\n\
};\n\
));
require.register("acuminous-yadda/lib/plugins/mocha/StepLevelPlugin.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var $ = require('../../Array');\n\
var Platform = require('../../Platform');\n\
var BasePlugin = require('./BasePlugin');\n\
\n\
module.exports.init = function(options) {\n\
\n\
    /* jslint shadow: true */\n\
    var options = options || {};\n\
    var platform = new Platform();\n\
    var container = options.container || platform.get_container();\n\
\n\
    var base_plugin = BasePlugin.create(options);\n\
\n\
    function scenarios(scenarios, iterator) {\n\
        $(scenarios).each(function(scenario) {\n\
            base_plugin.describe(scenario.title, scenario, iterator);\n\
        });\n\
    }\n\
\n\
    function steps(steps, iterator) {\n\
\n\
        var abort = false;\n\
\n\
        $(steps).each(function(step) {\n\
            var stepFn = iterator.length == 1 ? step_sync : step_async;\n\
            stepFn(step, iterator);\n\
        });\n\
\n\
        function step_async(step, iterator) {\n\
            base_plugin.it_async(step, step, function(step, done) {\n\
                if (abort) return done();\n\
                abort = true;\n\
                iterator(step, function(err) {\n\
                    if (err) return done(err);\n\
                    abort = false;\n\
                    done();\n\
                });\n\
            });\n\
        }\n\
\n\
        function step_sync(step, iterator) {\n\
            base_plugin.it_sync(step, step, function(step) {\n\
                if (abort) return;\n\
                abort = true;\n\
                iterator(step);\n\
                abort = false;\n\
            });\n\
        }\n\
    }\n\
\n\
    container.featureFiles = container.featureFile = base_plugin.featureFiles;\n\
    container.features = container.feature = base_plugin.features;\n\
    container.scenarios = container.scenario = scenarios;\n\
    container.steps = steps;\n\
};\n\
));
require.register("acuminous-yadda/lib/plugins/mocha/BasePlugin.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var English = require('../../localisation/English');\n\
var Platform = require('../../Platform');\n\
var FeatureFileParser = require('../../parsers/FeatureFileParser');\n\
var $ = require('../../Array');\n\
\n\
module.exports.create = function(options) {\n\
\n\
    /* jslint shadow: true */\n\
    var platform = new Platform();\n\
    var language = options.language || English;\n\
    var parser = options.parser || new FeatureFileParser(language);\n\
    var container = options.container || platform.get_container();\n\
\n\
    function featureFiles(files, iterator) {\n\
        $(files).each(function(file) {\n\
            features(parser.parse(file), iterator);\n\
        });\n\
    }\n\
\n\
    function features(features, iterator) {\n\
        $(features).each(function(feature) {\n\
            describe(feature.title, feature, iterator);\n\
        });\n\
    }\n\
\n\
    function describe(title, subject, iterator) {\n\
        var _describe = getDescribe(subject.annotations);\n\
        _describe(title, function() {\n\
            iterator(subject);\n\
        });\n\
    }\n\
\n\
    function it_async(title, subject, iterator) {\n\
        var _it = getIt(subject.annotations);\n\
        _it(title, function(done) {\n\
            iterator(subject, done);\n\
        });\n\
    }\n\
\n\
    function it_sync(title, subject, iterator) {\n\
        var _it = getIt(subject.annotations);\n\
        _it(title, function() {\n\
            iterator(subject);\n\
        });\n\
    }\n\
\n\
    function getIt(annotations, next) {\n\
        if (has_annotation(annotations, 'pending')) return container.xit;\n\
        if (has_annotation(annotations, 'only')) return container.it.only;\n\
        return container.it;\n\
    }\n\
\n\
    function getDescribe(annotations, next) {\n\
        if (has_annotation(annotations, 'pending')) return container.xdescribe;\n\
        if (has_annotation(annotations, 'only')) return container.describe.only;\n\
        return container.describe;\n\
    }\n\
\n\
    function has_annotation(annotations, name) {\n\
        var regexp = new RegExp('^' + language.localise(name) + '$');\n\
        for (var key in annotations) {\n\
            if (regexp.test(key)) return true;\n\
        }\n\
    }\n\
\n\
    return {\n\
        featureFiles: featureFiles,\n\
        features: features,\n\
        describe: describe,\n\
        it_async: it_async,\n\
        it_sync: it_sync\n\
    };\n\
};\n\
));
require.register("acuminous-yadda/lib/RegularExpression.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
 /* jslint node: true */\n\
 \"use strict\";\n\
\n\
var $ = require('./Array');\n\
\n\
// Wrapper for JavaScript Regular Expressions\n\
var RegularExpression = function(pattern_or_regexp) {\n\
\n\
    var groups_pattern = /(^|[^\\\\\\\\])\\(.*?\\)/g;\n\
    var sets_pattern = /(^|[^\\\\\\\\])\\[.*?\\]/g;\n\
    var repetitions_pattern = /(^|[^\\\\\\\\])\\{.*?\\}/g;\n\
    var regex_aliases_pattern = /(^|[^\\\\\\\\])\\\\./g;\n\
    var non_word_tokens_pattern = /[^\\w\\s]/g;\n\
    var regexp = new RegExp(pattern_or_regexp);\n\
\n\
    this.test = function(text) {\n\
        var result = regexp.test(text);\n\
        this.reset();\n\
        return result;\n\
    };\n\
\n\
    this.groups = function(text) {\n\
        var results = $();\n\
        var match = regexp.exec(text);\n\
        while (match) {\n\
            var groups = match.slice(1, match.length);\n\
            results.push(groups);\n\
            match = regexp.global && regexp.exec(text);\n\
        }\n\
        this.reset();\n\
        return results.flatten();\n\
    };\n\
\n\
    this.reset = function() {\n\
        regexp.lastIndex = 0;\n\
        return this;\n\
    };\n\
\n\
    this.without_expressions = function() {\n\
        return regexp.source.replace(groups_pattern, '$1')\n\
                            .replace(sets_pattern, '$1')\n\
                            .replace(repetitions_pattern, '$1')\n\
                            .replace(regex_aliases_pattern, '$1')\n\
                            .replace(non_word_tokens_pattern, '');\n\
    };\n\
\n\
    this.equals = function(other) {\n\
        return this.toString() == other.toString();\n\
    };\n\
\n\
    this.toString = function() {\n\
        return \"/\" + regexp.source + \"/\";\n\
    };\n\
};\n\
\n\
module.exports = RegularExpression;\n\
));
require.register("acuminous-yadda/lib/shims/index.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
\n\
/* jslint node: true */\n\
\"use strict\";\n\
\n\
var Platform = require('../Platform');\n\
\n\
module.exports = (function() {\n\
\n\
    var platform = new Platform();\n\
\n\
    var shims = {\n\
        node: function() {\n\
            return {\n\
                fs: require('fs'),\n\
                path: require('path'),\n\
                process: process\n\
            };\n\
        },\n\
        phantom: function() {\n\
            return {\n\
                fs: require('./phantom-fs'),\n\
                path: require('./phantom-path'),\n\
                process: require('./phantom-process')\n\
            };\n\
        },\n\
    };\n\
\n\
    function get_shim() {\n\
        if (platform.is_phantom()) return shims.phantom();\n\
        if (platform.is_node()) return shims.node();\n\
        return {};\n\
    }\n\
\n\
    return get_shim();\n\
})();\n\
));
require.register("acuminous-yadda/lib/Yadda.js", Function("exports, require, module",
"/*\n\
 * Copyright 2010 Acuminous Ltd / Energized Work Ltd\n\
 *\n\
 * Licensed under the Apache License, Version 2.0 (the \"License\");\n\
 * you may not use this file except in compliance with the License.\n\
 * You may obtain a copy of the License at\n\
 *\n\
 *     http://www.apache.org/licenses/LICENSE-2.0\n\
 *\n\
 * Unless required by applicable law or agreed to in writing, software\n\
 * distributed under the License is distributed on an \"AS IS\" BASIS,\n\
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n\
 * See the License for the specific language governing permissions and\n\
 * limitations under the License.\n\
 */\n\
/*jslint node: true */\n\
\"use strict\";\n\
\n\
var Interpreter = require('./Interpreter');\n\
var Context = require('./Context');\n\
var fn = require('./fn');\n\
\n\
var Yadda = function(libraries, interpreter_context) {\n\
\n\
    if (!(this instanceof Yadda)) {\n\
        return new Yadda(libraries, interpreter_context);\n\
    }\n\
\n\
    this.interpreter = new Interpreter(libraries);\n\
    var _this = this;\n\
\n\
    this.requires = function(libraries) {\n\
        this.interpreter.requires(libraries);\n\
        return this;\n\
    };\n\
\n\
    this.yadda = function(scenario, scenario_context, next) {\n\
        if (arguments.length === 0) return this;\n\
        if (arguments.length === 2 && fn.is_function(scenario_context)) return this.yadda(scenario, {}, scenario_context);\n\
        this.interpreter.validate(scenario);\n\
        this.interpreter.interpret(scenario, new Context().merge(interpreter_context).merge(scenario_context), next);\n\
    };\n\
\n\
    // Not everyone shares my sense of humour re the recursive api :(\n\
    // See https://github.com/acuminous/yadda/issues/111\n\
    this.run = this.yadda;\n\
\n\
    this.toString = function() {\n\
        return \"Yadda 0.11.5 Copyright 2010 Acuminous Ltd / Energized Work Ltd\";\n\
    };\n\
};\n\
\n\
module.exports = Yadda;\n\
));
require.register("yadda-component-browser-example/features/bottles.feature.js", Function("exports, require, module",
"module.exports = 'Feature: Mocha Asynchronous Example\\n\
\\n\
Scenario: A bottle falls from the wall\\n\
\\n\
    Given 100 green bottles are standing on the wall\\n\
    when 1 green bottle accidentally falls\\n\
    then there are 99 green bottles standing on the wall\\n\
\\n\
Scenario: No bottles are left\\n\
\\n\
    Given 1 green bottles are standing on the wall\\n\
    when 1 green bottle accidentally falls\\n\
    then there are 0 green bottles standing on the wall\\n\
\\n\
@Pending\\n\
Scenario: Bottles are reset\\n\
\\n\
    Given there are no green bottles\\n\
    when 5 minutes has elapsed\\n\
    then there are 100 green bottles standing on the wall\\n\
\\n\
Scenario: [N] bottles are standing on a wall\\n\
\\n\
    Given [N] green bottles are standing on the wall\\n\
    when 1 green bottle accidentally falls\\n\
    then there are [N-1] green bottles standing on the wall\\n\
\\n\
    Where:\\n\
        N   | N-1\\n\
        100 | 99\\n\
        99  | 98\\n\
        10  | 9';\n\
));
require.register("yadda-component-browser-example/bottles-library.js", Function("exports, require, module",
"/* jslint node: true */\n\
\"use strict\";\n\
\n\
var Yadda = require('yadda');\n\
var English = Yadda.localisation.English;\n\
var Dictionary = Yadda.Dictionary;\n\
var assert = require('chai').assert;\n\
\n\
module.exports = (function() {\n\
\n\
    var wall;\n\
    var dictionary = new Dictionary().define('NUM', /(\\d+)/);\n\
    var library = English.library(dictionary)\n\
\n\
    .given(\"$NUM green bottles are standing on the wall\", function(number_of_bottles, next) {\n\
        wall = new Wall(number_of_bottles);\n\
        next();\n\
    })\n\
\n\
    .when(\"$NUM green bottle accidentally falls\", function(number_of_falling_bottles, next) {\n\
        wall.fall(number_of_falling_bottles);\n\
        next();\n\
    })\n\
\n\
    .then(\"there (?:are|are still) $NUM green bottles standing on the wall\", function(number_of_bottles, next) {\n\
        assert.equal(number_of_bottles, wall.bottles);\n\
        next();\n\
    });\n\
\n\
    var Wall = function(bottles) {\n\
        this.bottles = bottles;\n\
        this.fall = function(n) {\n\
            this.bottles -= n;\n\
        };\n\
        this.returned = function() {\n\
            this.bottles++;\n\
        };\n\
    };\n\
\n\
    return library;\n\
})();\n\
));
require.alias("acuminous-yadda/lib/Array.js", "yadda-component-browser-example/deps/yadda/lib/Array.js");
require.alias("acuminous-yadda/lib/Competition.js", "yadda-component-browser-example/deps/yadda/lib/Competition.js");
require.alias("acuminous-yadda/lib/Context.js", "yadda-component-browser-example/deps/yadda/lib/Context.js");
require.alias("acuminous-yadda/lib/Dictionary.js", "yadda-component-browser-example/deps/yadda/lib/Dictionary.js");
require.alias("acuminous-yadda/lib/EventBus.js", "yadda-component-browser-example/deps/yadda/lib/EventBus.js");
require.alias("acuminous-yadda/lib/FileSearch.js", "yadda-component-browser-example/deps/yadda/lib/FileSearch.js");
require.alias("acuminous-yadda/lib/FeatureFileSearch.js", "yadda-component-browser-example/deps/yadda/lib/FeatureFileSearch.js");
require.alias("acuminous-yadda/lib/fn.js", "yadda-component-browser-example/deps/yadda/lib/fn.js");
require.alias("acuminous-yadda/lib/index.js", "yadda-component-browser-example/deps/yadda/lib/index.js");
require.alias("acuminous-yadda/lib/Interpreter.js", "yadda-component-browser-example/deps/yadda/lib/Interpreter.js");
require.alias("acuminous-yadda/lib/LevenshteinDistanceScore.js", "yadda-component-browser-example/deps/yadda/lib/LevenshteinDistanceScore.js");
require.alias("acuminous-yadda/lib/Library.js", "yadda-component-browser-example/deps/yadda/lib/Library.js");
require.alias("acuminous-yadda/lib/localisation/English.js", "yadda-component-browser-example/deps/yadda/lib/localisation/English.js");
require.alias("acuminous-yadda/lib/localisation/French.js", "yadda-component-browser-example/deps/yadda/lib/localisation/French.js");
require.alias("acuminous-yadda/lib/localisation/German.js", "yadda-component-browser-example/deps/yadda/lib/localisation/German.js");
require.alias("acuminous-yadda/lib/localisation/index.js", "yadda-component-browser-example/deps/yadda/lib/localisation/index.js");
require.alias("acuminous-yadda/lib/localisation/Language.js", "yadda-component-browser-example/deps/yadda/lib/localisation/Language.js");
require.alias("acuminous-yadda/lib/localisation/Pirate.js", "yadda-component-browser-example/deps/yadda/lib/localisation/Pirate.js");
require.alias("acuminous-yadda/lib/localisation/Polish.js", "yadda-component-browser-example/deps/yadda/lib/localisation/Polish.js");
require.alias("acuminous-yadda/lib/localisation/Russian.js", "yadda-component-browser-example/deps/yadda/lib/localisation/Russian.js");
require.alias("acuminous-yadda/lib/localisation/Spanish.js", "yadda-component-browser-example/deps/yadda/lib/localisation/Spanish.js");
require.alias("acuminous-yadda/lib/localisation/Norwegian.js", "yadda-component-browser-example/deps/yadda/lib/localisation/Norwegian.js");
require.alias("acuminous-yadda/lib/Macro.js", "yadda-component-browser-example/deps/yadda/lib/Macro.js");
require.alias("acuminous-yadda/lib/parsers/FeatureParser.js", "yadda-component-browser-example/deps/yadda/lib/parsers/FeatureParser.js");
require.alias("acuminous-yadda/lib/parsers/FeatureFileParser.js", "yadda-component-browser-example/deps/yadda/lib/parsers/FeatureFileParser.js");
require.alias("acuminous-yadda/lib/parsers/StepParser.js", "yadda-component-browser-example/deps/yadda/lib/parsers/StepParser.js");
require.alias("acuminous-yadda/lib/parsers/index.js", "yadda-component-browser-example/deps/yadda/lib/parsers/index.js");
require.alias("acuminous-yadda/lib/Platform.js", "yadda-component-browser-example/deps/yadda/lib/Platform.js");
require.alias("acuminous-yadda/lib/plugins/CasperPlugin.js", "yadda-component-browser-example/deps/yadda/lib/plugins/CasperPlugin.js");
require.alias("acuminous-yadda/lib/plugins/index.js", "yadda-component-browser-example/deps/yadda/lib/plugins/index.js");
require.alias("acuminous-yadda/lib/plugins/MochaPlugin.js", "yadda-component-browser-example/deps/yadda/lib/plugins/MochaPlugin.js");
require.alias("acuminous-yadda/lib/plugins/mocha/AsyncScenarioLevelPlugin.js", "yadda-component-browser-example/deps/yadda/lib/plugins/mocha/AsyncScenarioLevelPlugin.js");
require.alias("acuminous-yadda/lib/plugins/mocha/AsyncStepLevelPlugin.js", "yadda-component-browser-example/deps/yadda/lib/plugins/mocha/AsyncStepLevelPlugin.js");
require.alias("acuminous-yadda/lib/plugins/mocha/SyncScenarioLevelPlugin.js", "yadda-component-browser-example/deps/yadda/lib/plugins/mocha/SyncScenarioLevelPlugin.js");
require.alias("acuminous-yadda/lib/plugins/mocha/SyncStepLevelPlugin.js", "yadda-component-browser-example/deps/yadda/lib/plugins/mocha/SyncStepLevelPlugin.js");
require.alias("acuminous-yadda/lib/plugins/mocha/ScenarioLevelPlugin.js", "yadda-component-browser-example/deps/yadda/lib/plugins/mocha/ScenarioLevelPlugin.js");
require.alias("acuminous-yadda/lib/plugins/mocha/StepLevelPlugin.js", "yadda-component-browser-example/deps/yadda/lib/plugins/mocha/StepLevelPlugin.js");
require.alias("acuminous-yadda/lib/plugins/mocha/BasePlugin.js", "yadda-component-browser-example/deps/yadda/lib/plugins/mocha/BasePlugin.js");
require.alias("acuminous-yadda/lib/RegularExpression.js", "yadda-component-browser-example/deps/yadda/lib/RegularExpression.js");
require.alias("acuminous-yadda/lib/shims/index.js", "yadda-component-browser-example/deps/yadda/lib/shims/index.js");
require.alias("acuminous-yadda/lib/Yadda.js", "yadda-component-browser-example/deps/yadda/lib/Yadda.js");
require.alias("acuminous-yadda/lib/index.js", "yadda-component-browser-example/deps/yadda/index.js");
require.alias("acuminous-yadda/lib/index.js", "yadda/index.js");
require.alias("acuminous-yadda/lib/index.js", "acuminous-yadda/index.js");
